packages/firebase_messaging/firebase_messaging_platform_interface/lib/src/method_channel/method_channel_messaging.dart

```diff
@@ -34,78 +30,81 @@ void _firebaseMessagingCallbackDispatcher() {
 
   // This is where we handle background events from the native portion of the plugin.
   _channel.setMethodCallHandler((MethodCall call) async {
-    if (call.method == "MessagingBackground#onMessage") {
+    if (call.method == 'MessagingBackground#onMessage') {
       final CallbackHandle handle =
-          CallbackHandle.fromRawHandle(call.arguments["userCallbackHandle"]);
+          CallbackHandle.fromRawHandle(call.arguments['userCallbackHandle']);
 
       // PluginUtilities.getCallbackFromHandle performs a lookup based on the
       // callback handle and returns a tear-off of the original callback.
-      final Function closure = PluginUtilities.getCallbackFromHandle(handle);
+      final Function closure = PluginUtilities.getCallbackFromHandle(handle)!;
 
       if (closure == null) {
+        // ignore: avoid_print
         print('Fatal: could not find user callback');
         exit(-1);
       }
 
       try {
         Map<String, dynamic> messageMap =
-            Map<String, dynamic>.from(call.arguments["message"]);
+            Map<String, dynamic>.from(call.arguments['message']);
         final RemoteMessage remoteMessage = RemoteMessage.fromMap(messageMap);
         await closure(remoteMessage);
       } catch (e) {
+        // ignore: avoid_print
         print(
```

Comment:
not in the scope of the PR, but why are we printing here? Couldn't we let the exception bubble?

Reply:
cc @Salakar wdyt? I guess it's printing here since it's in an isolate?

Reply:
I can't recall the specifics but I think there was an issue where it'd get swallowed on the background isolate or something like that. Would have to try it out again. Tracking to revist this again on INVERTASE/FF-25.